x-pack/legacy/plugins/integrations_manager/public/components/assets_facet_group.tsx

```diff
@@ -56,8 +72,12 @@ export function AssetsFacetGroup({ assets }: { assets: AssetsGroupedByServiceByT
             </Header>
 
             <FacetGroup>
-              {entries(typeToParts).map(([type, parts]) => {
-                const iconType = AssetIcons[type];
+              {entries(filteredTypes).map(([type, parts]) => {
+                let iconType = null;
+                if (type in AssetIcons) {
+                  // only kibana assets have icons
+                  iconType = AssetIcons[type as KibanaAssetType];
```

Comment:
I'm struggling with this one. I wish the view could trust the data structure it's given and not need to know/repeat the logic for what's allowed.

IIUC, this breaks down when we add another type of asset which has icons, right? We can ship this under the banner of Good Enough for Now, but can you create a ticket to  come back to this (use data structure for assets/icons; don't duplicate logic in views)?

Reply:
If we have another type has that has icons I would change the name to AssetsWIthIcons or some such.  Since only Kibana does right now, I think this works.